File: e2e/playwright-python/context/task_context.py
import uuid
from data.task_data import TaskData
from data.task_response import TaskResponse
from data.task_status import TaskStatus
from data.task_priority import TaskPriority


class TaskContext:
    def __init__(
        self,
        id: str | None = None,
        title: str | None = None,
        description: str | None = None,
        status: TaskStatus = TaskStatus.TODO,
        priority: TaskPriority = TaskPriority.MEDIUM,
    ):
        self.id = id or str(uuid.uuid4())
        self.title = title or f"Task-{str(uuid.uuid4()).split('-')[0]}"
        self.description = description or "Automated test task description"
        self.status = status
        self.priority = priority

    def create_task_data(self) -> TaskData:
        return TaskData(
            title=self.title,
            description=self.description,
            status=self.status,
            priority=self.priority,
        )

    def create_task_response(self) -> TaskResponse:
        return TaskResponse(
            id=self.id,
            title=self.title,
            description=self.description,
            status=self.status,
            priority=self.priority,
        )

File: e2e/cucumber-playwright-typescript/context/TaskContext.ts
import { TaskPriority } from '@/data/TaskPriority';
import { TaskData } from '@/data/TaskData';
import { TaskRequest } from '@/data/TaskRequest';
import { TaskStatus } from '@/data/TaskStatus';
import { TaskResponse } from '@/data/TaskResponse';

type TaskContextParams = {
  id?: string;
  title?: string;
  description?: string;
  status?: TaskStatus;
  priority?: TaskPriority;
};

export class TaskContext {
  id: string;
  readonly title: string;
  readonly description: string;
  readonly status: TaskStatus;
  readonly priority: TaskPriority;

  constructor(overrides: TaskContextParams = {}) {
    this.id = overrides.id ?? crypto.randomUUID();
    this.title = overrides.title ?? `Task-${crypto.randomUUID().split('-')[0]}`;
    this.description = overrides.description ?? 'Automated test task description';
    this.status = overrides.status ?? TaskStatus.TODO;
    this.priority = overrides.priority ?? TaskPriority.MEDIUM;
  }

  createTaskRequest(): TaskRequest {
    return {
      title: this.title,
      description: this.description,
      status: this.status,
      priority: this.priority,
    };
  }

  createTaskData(): TaskData {
    return {
      title: this.title,
      description: this.description,
      status: this.status,
      priority: this.priority,
    };
  }

  createTaskResponse(): TaskResponse {
    return {
      id: this.id,
      title: this.title,
      description: this.description,
      status: this.status,
      priority: this.priority,
    };
  }
}
